src/core/qgsproject.cpp

```diff
@@ -820,6 +821,10 @@ bool QgsProject::addLayer( const QDomElement &layerElem, QList<QDomNode> &broken
   {
     mapLayer = new QgsRasterLayer;
   }
+  else if ( type == QLatin1String( "mesh" ) )
+  {
+    mapLayer = new QgsMeshLayer;
+  }
```

Comment:
Can we also have equivalent of encodedSource() for reading - e.g. decodedSource()

Reply:
done